src/time.cpp

```diff
@@ -2,17 +2,53 @@
 * 2012 by KjellKod.cc. This is PUBLIC DOMAIN to use at your own risk and comes
 * with no warranties. This code is yours to share, use and modify with no
 * strings attached and no restrictions or obligations.
-* 
+*
 * For more information see g3log/LICENSE or refer refer to http://unlicense.org
 * ============================================================================*/
 
 #include "g3log/time.hpp"
 
 #include <sstream>
 #include <string>
+#include <cstring>
+#include <cmath>
 #include <chrono>
 #include <cassert>
 #include <iomanip>
+#ifdef __MACH__
+#include <sys/time.h>
+#endif
+
+
+namespace {
+   const size_t kMsDigitsToCut = 1000000; // 3 digits left of 1000 000 000
+   const size_t kUsDigitsToCut = 1000; // 6 digits left
```

Comment:
think about the naming + what is needed to be here and what can be defined static namespace or in the function?